autocorpus/autocorpus.py

```diff
@@ -80,6 +90,45 @@ def __get_sections(self, soup, config):
 
         return handle_not_tables(config["sections"], soup)
 
+    @staticmethod
+    def __load_pdf_models():
+        global pdf_converter
+        if pdf_converter is None:
+            pdf_converter = PdfConverter(
+                artifact_dict=create_model_dict(),
+            )
+
+    @staticmethod
+    def __extract_pdf_content(
+        file_path: Path,
+    ) -> bool:
+        """Extracts content from a PDF file.
+
+        Args:
+            file_path (Path): Path to the PDF file.
+
+        Returns:
+            bool: success status of the extraction process.
+        """
+        bioc_text, bioc_tables = None, None
+        global pdf_converter
+        Autocorpus.__load_pdf_models()
+        if pdf_converter:
+            # extract text from PDF
+            rendered = pdf_converter(str(file_path))
+            text, _, _ = text_from_rendered(rendered)
+            # seperate text and tables
+            text, tables = extract_table_from_pdf_text(text)
+            # format data for BioC
+            bioc_text = BioCTextConverter(text, "pdf", str(file_path))
+            bioc_text.output_bioc_json(file_path)
+            bioc_tables = BioCTableConverter(tables, str(file_path))
+            bioc_tables.output_tables_json(file_path)
+            return True
+        else:
+            logger.error("PDF converter not initialized.")
+            return False
```

Comment:
`__load_pdf_models` should never silently fail, right? In which case you don't need this check:

```suggestion
        # extract text from PDF
        rendered = pdf_converter(str(file_path))
        text, _, _ = text_from_rendered(rendered)
        # separate text and tables
        text, tables = extract_table_from_pdf_text(text)
        # format data for BioC
        bioc_text = BioCTextConverter(text, "pdf", str(file_path))
        bioc_text.output_bioc_json(file_path)
        bioc_tables = BioCTableConverter(tables, str(file_path))
        bioc_tables.output_tables_json(file_path)
```

Reply:
Its possible, though I imagine the module itself should raise an error e.g. if the model loading fails, likely due to network problems during download or memory when loading them. I'll remove the check though, if an exception is thrown an if/else isn't going to catch it. Maybe a try/catch around the loader call instead?

Reply:
Good point... I'd forgotten that we probably want to recover and carry on processing other files even if the PDF converter refuses to load. Ignore my suggestion. Yes, a try/except seems like the right solution.

You could reorder for readability though:

```suggestion
        if not pdf_converter:
            logger.error("PDF converter not initialized.")
            return False

        # extract text from PDF
        rendered = pdf_converter(str(file_path))
        text, _, _ = text_from_rendered(rendered)
        # separate text and tables
        text, tables = extract_table_from_pdf_text(text)
        # format data for BioC
        bioc_text = BioCTextConverter(text, "pdf", str(file_path))
        bioc_text.output_bioc_json(file_path)
        bioc_tables = BioCTableConverter(tables, str(file_path))
        bioc_tables.output_tables_json(file_path)
        return True
```